tensorflow/examples/image_retraining/retrain.py

```diff
@@ -195,7 +197,7 @@ def create_image_lists(image_dir, testing_percentage, validation_percentage):
     if is_root_dir:
       is_root_dir = False
       continue
-    extensions = ['jpg', 'jpeg', 'JPG', 'JPEG']
+    extensions = ['jpg', 'jpeg', 'JPG', 'JPEG', 'gif', 'png']
```

Comment:
what about GIF and PNG (caps)

Maybe share this list between here and below? Might also not be a bad idea to make it a dictionary mapping extension to color depth (https://en.wikipedia.org/wiki/Color_depth). That way you can handle the conversion more generally...